### PR TITLE
8.0 book weight calculation

### PR DESCRIPTION
**Pull request by JehoG**
Product Template Book Weight Calculation

Implementation of the maths to calculate the weight of a book.

The maths for the calculation are following:

a(x/2) + b

where: \* a = Weight of one page of the book (Height (m) \* Width (m) \* Weight of the paper (kg)) \* b = Number of pages of the book \* c = Weight of the cover
Contributors

```
Jordi RIERA (jordi.riera@savoirfairelinux.com)
Bruno JOLIVEAU (bruno.joliveau@savoirfairelinux.com)
Guillaume AUGER (guillaume.auger@savoirfairelinux.com)
```

More information

Module developed and tested with Odoo version 8.0. For questions, please contact our support services (support@savoirfairelinux.com)

### CODE REVIEW

File: product_template_book_weight_calculation/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{
+    'name': 'Product Template Book Weight Calculation',
+    'version': '0.1',
```
Comment (foutoucour) [inline product_template_book_weight_calculation/__openerp__.py]: Minor: As the module seems not to be in beta, I would suggest to set the version to 1.0

**Comment by foutoucour** [comment]
:+1:

File: product_template_book_weight_calculation/README.rst [hunk 2]
```diff
@@ -0,0 +1,29 @@
+Product Template Book Weight Calculation
```
Comment (max3903) [inline product_template_book_weight_calculation/README.rst:1]: Please use the template from maintainers-tools.

**Comment by JehoG** [comment]
This travis error seems odd to me. Someone knows something about that?

**Comment by pedrobaeza** [comment]
Sometimes Travis fails. I have restarted build and now is OK.

**Comment by pedrobaeza** [comment]
As I can see, this is only an utility that can be called from any other module, but the module itself doesn't add any model nor view, and the inherited class product.template is arbitrary, becase all the needed data is passed as arguments, so there's no need to put it there, but it can be a regular Python class.

This way, I would make these questions:
- Is the module really needed as an isolated one? It's very short in functionality, and this can be easily taken as a compute method of a field.
- Module name is not very indicated and it's very long. In case the module continues as is (but I don't recommend it), a better name is _base_book_weight_, as this is only a base module that needs another ones to unfold its functionality.

**Comment by foutoucour** [comment]
Hey @pedrobaeza,

The maths could be isolated as you recommend but would not necessary make sense. The maths are built to calculate the weight of a book, which seems to be a product in Odoo. In that sense, the method has been embed.
As it is a static method, it can be anyway easily imported to be used.

Regarding the name, we follow your advices form #2 and put a name representing its usage. We thought about short name as you advice here, but it seemed to be too generic.

regards,
Jordi

**Comment by pedrobaeza** [comment]
Check new pep8 error:

```
./product_template_book_weight_calculation/product_template_book_weight_calculation.py:24:1: E402 module level import not at top of file
```

and the explanation is :+1:

**Comment by max3903** [comment]
:+1:

**Pull request merged**